### PR TITLE
Add infer static code analysis to CI

### LINKED ISSUE #289: Add infer static code analysis to CI
Infer: https://fbinfer.com
https://github.com/facebook/infer

### PR DESCRIPTION
**Pull request by avinal**
Fixes #289

## Proposed changes 

- Adds infer
- remove parallel from builds
- Add name to code formatting workflow

----
Signed-off-by: Avinal Kumar <avinal.xlvii@gmail.com>

### CODE REVIEW

**Comment by pktiuk** [comment]
Will reports from infer be also available in [security page](https://github.com/AntiMicroX/antimicrox/security/code-scanning)?

**Comment by avinal** [comment]
> Will reports from infer be also available in [security page](https://github.com/AntiMicroX/antimicrox/security/code-scanning)?

No, security page requires reports in SARIF format, infer only emits JSON and text.

**Review by pktiuk — CHANGES_REQUESTED** [review]

File: .github/workflows/static-code-analysis.yml [hunk 1]
```diff
@@ -38,7 +38,35 @@ jobs:
               run: cmake -GNinja -DCMAKE_BUILD_TYPE=Debug -B ${{ github.workspace }}/build
 
             - name: Build AntiMicroX
-              run: cmake --build ${{ github.workspace }}/build --parallel 4
+              run: cmake --build ${{ github.workspace }}/build
 
             - name: Run CodeQL
               uses: github/codeql-action/analyze@v1
+
+    infer-analysis:
```
[the remark this answers is not in the document]
Comment (avinal) [reply .github/workflows/static-code-analysis.yml:46]: ok I will add that too

[the remark this answers is not in the document]
Comment (pktiuk) [reply .github/workflows/static-code-analysis.yml:46]: Remember to resolve this conversation and Re-request review when you apply changes

[the remark this answers is not in the document]
Comment (avinal) [reply .github/workflows/static-code-analysis.yml:46]: Sure, I will complete it by the weekend.

**Review by pktiuk — APPROVED** [review]

**Pull request merged**